src/SubtitlesParser/index.js

```diff
@@ -0,0 +1,189 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2020 Metrological
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Log } from '../LightningSdkPlugins'
+
+export default class SubtitlesParser {
+  // @ params url: subtitle file URL
+  // @ customParser: a customParser to use instead of default parser of the plugin
+  // @ parseOptions.removeSubtitleTextStyles: remove subtitle textstyles possible value true or false
+  // @return parsed subtitles as list of objects
+  // also stores parsed data
+  static fetchAndParseSubs(
+    url,
+    customParser = false,
+    parseOptions = { removeSubtitleTextStyles: true }
+  ) {
+    const _url = new URL(url)
+    if (!((_url.protocol === 'https:' || _url.protocol === 'http:') && _url.hostname)) {
+      Log.info('Invalid subtitle Url')
+      return Promise.reject(new Error('Invalid URL'))
+    }
```

Comment:
This will fail if a url of `bla` is passed in. Line 33 will throw an `Uncaught TypeError: Failed to construct 'URL': Invalid URL`. So, I would maybe use a try/catch mechanism instead;

```js
let _url;

try {
	_url = new URL(url);	// This will throw when an invalid url is passed
} catch (e) {
	Log.info('Invalid URL');
	return;
}
```